bofa_data.php

```diff
@@ -2185,6 +2185,1733 @@
+"G imp" => [['Pandamonium Slums', '95', 7], ['The Dark Elbow of the Woods', '95', 3], ['The Dark Heart of the Woods', '95', 3]],
+"Hellion" => [['Pandamonium Slums', '95', 7], ['The Dark Neck of the Woods', '95', 3]],
+"L imp" => [['Pandamonium Slums', '95', 7], ['The Dark Elbow of the Woods', '95', 3]],
+"P imp" => [['Pandamonium Slums', '95', 7], ['The Dark Heart of the Woods', '95', 3], ['The Dark Neck of the Woods', '95', 3]],
+"W imp" => [['Pandamonium Slums', '95', 7], ['The Dark Neck of the Woods', '95', 3], ['The Road to the White Citadel', '100', 1, 'special']],
+"Hypnotist of Hey Deze" => [['Pandamonium Slums', '95', 7, 'UR']],
+"filthy pirate" => [['Pirates of the Garbage Barges', '100', 4]],
+"fishy pirate" => [['Pirates of the Garbage Barges', '100', 4]],
+"funky pirate" => [['Pirates of the Garbage Barges', '100', 4]],
+"flashy pirate" => [['Pirates of the Garbage Barges', '100', 4]],
+"former inmate" => [['Porkham Asylum', '100', 3]],
+"former guard" => [['Porkham Asylum', '100', 3]],
+"The Author" => [['Porkham Asylum', '100', 3, 'special']],
+"Castle in the Clouds in the Sky" => [["Professor Jacking's Huge-A-Ma-tron", '-1', 3]],
+"Fearsome Wacken" => [["Professor Jacking's Huge-A-Ma-tron", '-1', 3, 'special']],
+"loose coalition of yetis, snowmen, and goats" => [["Professor Jacking's Huge-A-Ma-tron", '-1', 3]],
+"Spookyraven Manor" => [["Professor Jacking's Huge-A-Ma-tron", '-1', 3]],
+"The Whole Kingdom" => [["Professor Jacking's Huge-A-Ma-tron", '-1', 3, 'special']],
+"cruel dust mote" => [["Professor Jacking's Small-O-Fier", '-1', 3]],
+"funk particle" => [["Professor Jacking's Small-O-Fier", '-1', 3]],
+"jungle scabie" => [["Professor Jacking's Small-O-Fier", '-1', 3, 'special']],
+"malevolent magnetic field" => [["Professor Jacking's Small-O-Fier", '-1', 3]],
+"smooth jazz scabie" => [["Professor Jacking's Small-O-Fier", '-1', 3, 'special']],
+"ancient ghost" => [["Regret Man's Level", '100', 3]],
+"contemplative ghost" => [["Regret Man's Level", '100', 3]],
+"lovesick ghost" => [["Regret Man's Level", '100', 3]],
+"Regret Man" => [["Regret Man's Level", '100', 3, 'special']],
+"bugbear scientist" => [['Science Lab', '100', 2]],
+"spiderbugbear" => [['Science Lab', '100', 2]],
+"7-Foot Dwarf Replicant" => [['Seaside Megalopolis', '-1', 8]],
+"Bangyomaman Warrior" => [['Seaside Megalopolis', '-1', 8]],
+"cyborg policeman" => [['Seaside Megalopolis', '-1', 8]],
+"\"Handyman\" Jay Android" => [['Seaside Megalopolis', '-1', 8]],
+"liquid metal robot" => [['Seaside Megalopolis', '-1', 8]],
+"obese tourist" => [['Seaside Megalopolis', '-1', 8]],
+"Space Marine" => [['Seaside Megalopolis', '-1', 8]],
+"terrifying robot" => [['Seaside Megalopolis', '-1', 8]],
+"shadow bat" => [['Shadow Rift', '100', 12], ['Shadow Rift (Spookyraven Manor Third Floor)', '100', 3], ['Shadow Rift (The Ancient Buried Pyramid)', '100', 3], ['Shadow Rift (The Castle in the Clouds in the Sky)', '100', 3], ['Shadow Rift (The Nearby Plains)', '100', 3]],
```

Comment:
was the first element here (the [shadow rift] thing) done to abstract this into a summary zone or is this something that was generated by default through the logic that refactored combats.txt? it looks like it existed in combats.txt but we may need bespoke handling for rift guys at some point

Reply:
It was generated from combats.txt, I did not add any manual changes.